doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1417,77 +1412,57 @@ Managing the local context
    .. exn:: The variable @ident is already defined.
       :undocumented:
 
-   .. tacv:: set (@ident := @term) in @goal_occurrences
-
-      This notation allows specifying which occurrences of :token:`term` have
-      to be substituted in the context. The :n:`in @goal_occurrences` clause
-      is an occurrence clause whose syntax and behavior are described in
-      :ref:`goal occurrences <occurrenceclauses>`.
-
-   .. tacv:: set (@ident {* @binder } := @term) {? in @goal_occurrences }
-
-      This is equivalent to :n:`set (@ident := fun {* @binder } => @term) {? in @goal_occurrences }`.
-
-   .. tacv:: set @term {? in @goal_occurrences }
+   .. tacn:: eset @bindings_with_parameters {? @occurrences }
+             eset @one_term {? @as_name } {? @occurrences }
+      :name: eset; _
 
-      This behaves as :n:`set (@ident := @term) {? in @goal_occurrences }`
-      but :token:`ident` is generated by Coq.
+      .. insertprodn as_name as_name
 
-   .. tacv:: eset (@ident {* @binder } := @term) {? in @goal_occurrences }
-             eset @term {? in @goal_occurrences }
-      :name: eset; _
+      .. prodn::
+         as_name ::= as @ident
 
-      While the different variants of :tacn:`set` expect that no existential
-      variables are generated by the tactic, :tacn:`eset` removes this
-      constraint. In practice, this is relevant only when :tacn:`eset` is
+      Similar to :tacn:`set`, but generates an existential variable when
+      a variable can't be instantiated instead of failing.
```

Comment:
Here the wording is a little confusing. I think it should be `existential variable can't be instantiated` rather than just variable, since what we are thinking about is a hole like `_`.

Perhaps saying "incomplete term" would be more accurate.

Reply:
Can you suggest a complete sentence?  Probably should include holes in the explanation.

Reply:
```
Similar to :tacn:`set`, but generates an existential variable when
      a variable can't be instantiated instead of failing.
```
to
```
Similar to :tacn:`set`, but leaves existential variables if they cannot be instantiated instead of failing.
```

Reply:
How about something like:

```
      Similar to :tacn:`set`, but instead of failing because of uninstantiated
      variables, generates existential variables for them.
```

I'd prefer "unbound" over "uninstantiated" but we have messages such as "Unable to find an instance for the variable m."  Maybe that's not the best wording, WDYT?

Also we two distinct concepts here: "variable" and "existential variable".  Readers may miss that detail.

Reply:
Bound has a distinct meaning when it comes to variables however. It is about the name and scope of a variable. I don't think using bound is a good idea.

I'm in favor of perhaps talking about existential variables as "holes". But this will have to be adopted more universally in the refman.

Reply:
Holes (`_`) in terms are distinct from existential variables (`?x`) so we shouldn't conflate the two concepts.  Holes can't appear in the context.  Muddying things a bit is that `_` is used for several things.